lib/fastlane/actions/tryouts.rb

```diff
@@ -0,0 +1,124 @@
+module Fastlane
+  module Actions
+    module SharedValues
+      # Contains all the data returned from the Tryouts API. See http://tryouts.readthedocs.org/en/latest/releases.html#create-release
+      TRYOUTS_BUILD_INFORMATION = :TRYOUTS_BUILD_INFORMATION
+    end
+    class TryoutsAction < Action
+
+      TRYOUTS_API_BUILD_RELEASE_TEMPLATE = "https://api.tryouts.io/v1/applications/%s/releases/"
+
+      def self.run(params)
+        Helper.log.info 'Upload to Tryouts has been started. This may take some time.'.green
+
+        response = self.upload_build(params)
+
+        case response.status
+        when 200...300
+          Actions.lane_context[SharedValues::TRYOUTS_BUILD_INFORMATION] = response.body
+          Helper.log.info 'Build successfully uploaded to Tryouts!'.green
+          Helper.log.info "Release download url: #{response.body['download_url']}" if response.body["download_url"]
+        else
+          raise "Error when trying to upload build file to Tryouts: #{response.body}".red
+        end
+      end
+
+      def self.upload_build(params)
+        require 'faraday'
+        require 'faraday_middleware'
+
+        url = TRYOUTS_API_BUILD_RELEASE_TEMPLATE % params[:app_identifier]
+        connection = Faraday.new(url) do |builder|
+          builder.request :multipart
+          builder.request :url_encoded
+          builder.response :json, content_type: /\bjson$/
+          builder.use FaradayMiddleware::FollowRedirects
+          builder.adapter :net_http
+        end
+
+        options = {}
+        options[:build] = Faraday::UploadIO.new(params[:build_file], 'application/octet-stream')
+
+        if params[:notes_path]
+          options[:notes] = File.read(params[:notes_path])
+        else
+          options[:notes] = params[:notes] if params[:notes]
+        end
+
+        options[:notify] = params[:notify]
```

Comment:
The user might just use the action using `tryouts(notify: 1)` (passing an integer)